### PR TITLE
User can sign up

### PR DESCRIPTION
**Pull request by GergKllai1**
Creates user can sign up feature:

- Initializes devise with all views
- Adds address, phone and name to database
- Creates validations for these attributes
- Adds these attributes to the sign up form
- Creates welcome index to display sign up link
- Adds unit tests for Factory, and feature tests for happy and sad path

PT:
https://www.pivotaltracker.com/n/projects/2227327

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,290 @@
+# frozen_string_literal: true
+
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
```
Comment (oliverochman) [inline config/initializers/devise.rb]: clean this file up

File: app/controllers/welcome_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class WelcomeController < ApplicationController
```
Comment (oliverochman) [inline app/controllers/welcome_controller.rb]: Do you need this controller for anything?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/welcome_controller.rb (hunk 2 above)
Comment (GergKllai1) [reply app/controllers/welcome_controller.rb]: No, removed it

File: config/initializers/devise.rb (hunk 1 above)
Comment (GergKllai1) [reply config/initializers/devise.rb]: done

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :name, presence: true
```
Comment (oliverochman) [inline app/models/user.rb]: validates_presence_of :name, :address, :phone

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+RSpec.describe User, type: :model do
+
+  describe 'db table' do
+    it { is_expected.to have_db_column :name }
+    it { is_expected.to have_db_column :email }
+    it { is_expected.to have_db_column :address }
+    it { is_expected.to have_db_column :phone }
+    it { is_expected.to have_db_column :encrypted_password }
+  end
+
```
Comment (oliverochman) [inline spec/models/user_spec.rb:12]: You need to test for the validations that you have on the user model

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by oliverochman — APPROVED** [review]
LGTM!

**Pull request merged**